### PR TITLE
Add more docs

### PR DESCRIPTION
**Pull request by markstory**
Import the docs from the book and add some additional content.

### CODE REVIEW

File: src/functions.php [hunk 1]
```diff
@@ -17,6 +17,7 @@
 
 use Cake\ORM\Table;
 use Cake\ORM\TableRegistry;
+use Cake\Routing\Router;
```
Comment (ADmad) [inline src/functions.php]: Umm..why?

**Comment by ADmad** [comment]
.gitattributes should be updated to ignore the new files from zip balls.

File: src/functions.php (hunk 1 above)
Comment (markstory) [reply src/functions.php]: Cause I'm dumb. I was thinking of adding a url() function and then changed my mind.

Comment (ADmad) [reply src/functions.php]: I have at times contemplated having a `url()` function in core itself for regular use :)

**Pull request merged**